Assets/Scripts/Dashboard/Button.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using Anivision.Core;
+using Anivision.PlayerInteraction;
+
+public class Button : MonoBehaviour
+{
+    public Color defaultColor;
+    public Color hoverColor;
+    [HideInInspector] public UnityEvent onClick;
+
+    private InputManager _inputManager;
+    private HapticsController _hapticsController;
+    private TeleportController _teleportController;
+
+    private void OnEnable()
+    {
+        _inputManager = InputManager.Instance;
+        if (_inputManager == null) throw new System.Exception("Must have an input manager script in the scene");
+
+        _hapticsController = HapticsController.Instance;
+        _teleportController = FindObjectOfType(typeof(TeleportController)) as TeleportController;
```

Comment:
`TeleportController` is a singleton right now, so you can just do `TeleportController.Instance`